plugins/jobs/src/js/components/form/RunConfigFormSection.tsx

```diff
@@ -0,0 +1,424 @@
+import * as React from "react";
+import { Trans } from "@lingui/macro";
+
+import AddButton from "#SRC/js/components/form/AddButton";
+import DeleteRowButton from "#SRC/js/components/form/DeleteRowButton";
+import FieldAutofocus from "#SRC/js/components/form/FieldAutofocus";
+import FieldHelp from "#SRC/js/components/form/FieldHelp";
+import FieldInput from "#SRC/js/components/form/FieldInput";
+import FieldLabel from "#SRC/js/components/form/FieldLabel";
+import FormGroup from "#SRC/js/components/form/FormGroup";
+import FormGroupContainer from "#SRC/js/components/form/FormGroupContainer";
+import FormGroupHeading from "#SRC/js/components/form/FormGroupHeading";
+import FormGroupHeadingContent from "#SRC/js/components/form/FormGroupHeadingContent";
+import FormRow from "#SRC/js/components/form/FormRow";
+import InfoTooltipIcon from "#SRC/js/components/form/InfoTooltipIcon";
+import classNames from "classnames";
+import { FormOutput, FormError, RestartPolicy } from "./helpers/JobFormData";
+import { Tooltip } from "reactjs-components";
+
+interface RunConfigSectionProps {
+  formData: FormOutput;
+  errors: FormError[];
+  showErrors: boolean;
+  onRemoveItem: (path: string, index: number) => void;
+  onAddItem: (path: string) => void;
+}
+
+class RunConfigFormSection extends React.Component<RunConfigSectionProps> {
+  constructor(props: RunConfigSectionProps) {
+    super(props);
+  }
+
+  render() {
+    const { formData, onAddItem, onRemoveItem } = this.props;
+    const showErrors = false;
+    const labels = formData.labels || [];
+    const artifacts = formData.artifacts || [];
+
+    try {
```

Comment:
Why are we using a `try` here? Are we sure about the impacts of missing out on the code optimisation which v8 might not do because of the try catch block or has this been changed in v8?

Reply:
because i don't trust the validators and reducers to always bring "good" data here, when someone enters weird stuff via JSON. implementing an approach i'd trust would've been out of scope here.

AFAIK only code within a try-catch is affected. also we have a componentFromStream that will be wrapped in a try-catch due to every observable being wrapper, right? i guess the implications for one render function are not big enough to continue this discussion (unless my understanding is wrong and the whole app is now affected for some reason).